diff --git a/src/App.css b/src/App.css
index e817bcd..4982210 100644
--- a/src/App.css
+++ b/src/App.css
@@ -1,15 +1,25 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
-*{
+* {
   box-sizing: border-box;
   margin: 0;
   padding: 0;
 }
 .Ael {
-    width: 100%;
-    height: 100vh;
-    padding: 25px;
-    display: flex;
-    align-items: center;
-}
\ No newline at end of file
+  width: 100%;
+  height: 100vh;
+  padding: 25px;
+  display: flex;
+  align-items: center;
+}
+@media screen and (max-width: 1000px) {
+  .Ael {
+    height: auto;
+  }
+}
+@media screen and (max-width: 520px) {
+  .Ael {
+    margin-bottom: 100px;
+  }
+}
diff --git a/src/App.js b/src/App.js
index 881060c..23762bc 100644
--- a/src/App.js
+++ b/src/App.js
@@ -1,4 +1,4 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { Counter } from "./features/counter/Counter";
 import SideFIlters from "./components/filters/SideFIlters";
 import DisplayInfo from "./components/maincomponents/DisplayInfo";
@@ -10,11 +10,21 @@ import Kp4 from "./components/maincomponents/Kp4";
 import "./App.css";
 import "./style/mainComp.css";
 import "./index.css";
+import { FaSearch, FaBars } from "react-icons/fa";
 
 function App() {
+  const [openFilters, setFIltetrs] = useState(true);
+
   return (
     <div className="Ael">
-      <SideFIlters />
+      <FaBars
+        className="hamburger"
+        onClick={() => {
+          setFIltetrs(() => !openFilters);
+        }}
+      />
+      <SideFIlters setFIltetrs={setFIltetrs} openFilters={openFilters} />
+
       <div className="main">
         <div className="infogrid">
           <div>
diff --git a/src/components/filters/Select.jsx b/src/components/filters/Select.jsx
index 855eccd..71c3097 100644
--- a/src/components/filters/Select.jsx
+++ b/src/components/filters/Select.jsx
@@ -3,20 +3,19 @@ import { BiChevronDown } from "react-icons/bi";
 import { AiOutlineSearch } from "react-icons/ai";
 import { FaSortDown } from "react-icons/fa";
 
-const Selector = ({ title }) => {
-  const [countries, setCountries] = useState(null);
+const Selector = ({ title, api }) => {
+  const [filtersOption, setFilters] = useState(null);
   const [inputValue, setInputValue] = useState("");
   const [selected, setSelected] = useState("");
   const [open, setOpen] = useState(false);
   console.log(selected);
 
   useEffect(() => {
-    fetch("https://restcountries.com/v2/all?fields=name")
+    fetch(api)
       .then((res) => res.json())
       .then((data) => {
-       
-          setCountries(data.splice(3));
-       
+        console.log(data,'data');
+        setFilters(data.message);
       });
   }, []);
   return (
@@ -24,7 +23,7 @@ const Selector = ({ title }) => {
       <div
         onClick={() => setOpen(!open)}
         className={`bg-gray-500 text-white w-full p-2 flex items-center justify-between rounded ${
-          !selected && "text-gray-700"
+          !selected && "text-gray-700 text-white"
         }`}
       >
         {selected
@@ -32,35 +31,38 @@ const Selector = ({ title }) => {
             ? selected?.substring(0, 25) + "..."
             : selected
           : title}
-        <FaSortDown size={20} className={`${open && "rotate-180"}`} />
+        <FaSortDown
+          size={20}
+          className={`text-white ${open && "rotate-180"}`}
+        />
       </div>
       <ul
         className={`bg-white mt-2 w-11/12 rounded-md border-2 border-gray-400  mx-auto overflow-y-auto overflow-x-hidden ${
           open ? "max-h-60" : "max-h-0"
         } `}
       >
-        {countries?.map((country) => (
+        {filtersOption?.map((option) => (
           <li
-            key={country?.name}
+            key={option?.name}
             className={`p-2 text-sm hover:bg-sky-600 hover:text-white
             ${
-              country?.name?.toLowerCase() === selected?.toLowerCase() &&
-              "bg-sky-600 text-white"
+              option?.name?.toLowerCase() === selected?.toLowerCase() &&
+              "bg-sky-600 "
             }
             ${
-              country?.name?.toLowerCase().startsWith(inputValue)
+              option?.name?.toLowerCase().startsWith(inputValue)
                 ? "block"
                 : "hidden"
             }`}
             onClick={() => {
-              if (country?.name?.toLowerCase() !== selected.toLowerCase()) {
-                setSelected(country?.name);
+              if (option?.name?.toLowerCase() !== selected.toLowerCase()) {
+                setSelected(option?.name);
                 setOpen(false);
                 setInputValue("");
               }
             }}
           >
-            {country?.name}
+            {option?.name}
           </li>
         ))}
       </ul>
diff --git a/src/components/filters/SideFIlters.jsx b/src/components/filters/SideFIlters.jsx
index 4cb304f..13dcf8a 100644
--- a/src/components/filters/SideFIlters.jsx
+++ b/src/components/filters/SideFIlters.jsx
@@ -1,11 +1,19 @@
-import React from 'react';
-import "../../style/sidefilters.css"
-import Selector from './Select';
-import { FaSearch } from "react-icons/fa";
+import React ,{useState,useEffect}from "react";
+import "../../style/sidefilters.css";
+import Selector from "./Select";
+import { FaSearch,FaBars } from "react-icons/fa";
 
-function SideFIlters() {
+function SideFIlters({ openFilters, setFIltetrs }) {
+  // useEffect(() => {
+  //   console.log(setFilters, "lo");
+  // }, [setFilters]);
+  const gradeApi='http://13.230.65.59:8000/coal/grade'
+  const originApi='http://13.230.65.59:8000/coal/origin'
+  const portApi='http://13.230.65.59:8000/coal/port'
   return (
-    <div className="sidefilters ">
+    <div
+      className={` ${openFilters ? "showfilter" : "hidefilters"} sidefilters`}
+    >
       <div className="filterhead">
         <h1>Filters</h1>
         <h1>Clear Filters</h1>
@@ -20,12 +28,11 @@ function SideFIlters() {
         />
         <FaSearch className="icon" />
       </div>
-
-      <Selector title="Port"/>
-      <Selector title="Grade" />
-      <Selector title="Origin" />
+      <Selector title="Port" api={portApi} />
+      <Selector title="Grade" api={gradeApi} />
+      <Selector title="Origin" api={originApi} />
     </div>
   );
 }
 
-export default SideFIlters;
\ No newline at end of file
+export default SideFIlters;
diff --git a/src/components/maincomponents/PriceForm.jsx b/src/components/maincomponents/PriceForm.jsx
index 234860f..09a07b7 100644
--- a/src/components/maincomponents/PriceForm.jsx
+++ b/src/components/maincomponents/PriceForm.jsx
@@ -1,27 +1,61 @@
-import React from 'react'
+import React, { useEffect, useState } from "react";
 
 function PriceForm() {
+  const [priceValues, setValues] = useState({
+    competitor: "",
+    baseprice: "",
+    indexprice: "",
+  });
+  function changeHandler(e) {
+    let value = e.target.value;
+    let name = e.target.name;
+    setValues({ ...priceValues, [name]: value });
+  }
+  function submit() {
+    console.log(priceValues, "priceValues");
+    setValues({
+      competitor: "",
+      baseprice: "",
+      indexprice: "",
+    });
+  }
+
   return (
     <div className="priceform">
       <form>
         <div>
           <h1>Competitor Price</h1>
-          <input type="text" />
+          <input
+            type="text"
+            name="competitor"
+            value={priceValues.competitor}
+            onChange={(e) => changeHandler(e)}
+          />
         </div>
         <div>
           <h1>Base Price</h1>
-          <input type="text" />
+          <input
+            type="text"
+            name="baseprice"
+            value={priceValues.baseprice}
+            onChange={(e) => changeHandler(e)}
+          />
         </div>
         <div>
           <h1>Index Price</h1>
-          <input type="text" />
+          <input
+            type="text"
+            name="indexprice"
+            value={priceValues.indexprice}
+            onChange={(e) => changeHandler(e)}
+          />
         </div>
       </form>
       <div className="btn-con">
-        <button>Submit</button>
+        <button onClick={submit}>Submit</button>
       </div>
     </div>
   );
 }
 
-export default PriceForm;
\ No newline at end of file
+export default PriceForm;
diff --git a/src/style/mainComp.css b/src/style/mainComp.css
index 455a2fc..533dc69 100644
--- a/src/style/mainComp.css
+++ b/src/style/mainComp.css
@@ -25,6 +25,7 @@
 .priceform form div input {
   height: 5vh;
   border-radius: 3px;
+  padding: 0px 10px;
 }
 .priceform form div h1 {
   font-size: 25px;
@@ -133,3 +134,61 @@
   background-color: rgb(223, 222, 220);
   margin-top: 5px;
 }
+.kp1:hover {
+  background-color: rgb(193, 193, 193);
+}
+.kp2:hover {
+  background-color: rgb(193, 193, 193);
+}
+.kp3:hover {
+  background-color: rgb(193, 193, 193);
+}
+.sec-sec:hover {
+  background-color: rgb(193, 193, 193);
+}
+
+/*  responsiveness */
+@media screen and (max-width: 1200px) {
+  .main {
+    height: 100%;
+  }
+  .priceform form {
+    width: 100%;
+    display: grid;
+    grid-template-columns: repeat(1, 1fr);
+    height: 40vh;
+  }
+  .priceform form div input {
+    width: 80%;
+  }
+  .btn-con button {
+    bottom: 5px;
+  }
+}
+@media screen and (max-width: 540px) {
+  .main {
+    width: 100%;
+    z-index: 100;
+  }
+  .info {
+    display: grid;
+    grid-template-columns: repeat(1, 1fr);
+    height: auto;
+  }
+  .first-sec,
+  .sec-sec {
+    width: 100%;
+  }
+  .sec-sec {
+    height: 17vh;
+  }
+  .first-sec .total {
+    width: 100%;
+    height: 100%;
+    padding: 20px;
+    display: flex;
+    align-items: center;
+    text-align: center;
+    justify-content: space-between;
+  }
+}
diff --git a/src/style/sidefilters.css b/src/style/sidefilters.css
index 09a9fd1..c576baa 100644
--- a/src/style/sidefilters.css
+++ b/src/style/sidefilters.css
@@ -1,8 +1,9 @@
 .sidefilters {
   width: 17%;
-  height: 100%;
+  height: 95vh;
   background-color: rgb(223, 222, 220);
   border: 2px solid rgb(193, 186, 186);
+  overflow: scroll;
 }
 .filterhead {
   width: 90%;
@@ -13,21 +14,60 @@
   font-weight: bolder;
   font-size: 20px;
 }
-.search{
-    display: block;
-    position: relative;
+.search {
+  display: block;
+  position: relative;
 }
-.search input{
-    font-size: 19px;
-    color: rgb(121, 121, 115);
-    font-weight: 500;
+.search input {
+  font-size: 19px;
+  color: rgb(121, 121, 115);
+  font-weight: 500;
 }
-.search .icon{
-    position: absolute;
-    top: 28% ;
-    left: 10%;
-    font-size: 24px;
-    color: rgb(107, 107, 206);
+.search .icon {
+  position: absolute;
+  top: 28%;
+  left: 10%;
+  font-size: 24px;
+  color: rgb(107, 107, 206);
+}
+.hamburger {
+  position: absolute;
+  left: 7%;
+  top: 0.5%;
+  z-index: 200;
+  display: none;
 }
 
-
+@media screen and (max-width: 1000px) {
+  .sidefilters {
+    height: 105vh;
+  }
+}
+@media screen and (max-width: 820px) {
+  .sidefilters {
+    width: 30%;
+  }
+}
+@media screen and (max-width: 520px) {
+  .showfilter {
+    width: 50%;
+    position: fixed;
+    /* display: none; */
+    z-index: 200;
+    top: 2.6%;
+    left: 0%;
+    overflow: scroll;
+    padding-bottom:60px ;
+  }
+  .hidefilters{
+    width: 50%;
+    position: absolute;
+    display: none;
+    /* left: -40%; */
+    z-index: 200;
+  }
+  .hamburger{
+    display: block;
+    position: fixed;
+  }
+}
