diff --git a/package-lock.json b/package-lock.json
index bb57540..dde1801 100644
--- a/package-lock.json
+++ b/package-lock.json
@@ -22,6 +22,7 @@
         "react-router-dom": "^6.4.3",
         "react-scripts": "^5.0.1",
         "react-toastify": "^9.1.1",
+        "source-map-resolve": "^0.6.0",
         "web-vitals": "^2.1.4"
       },
       "devDependencies": {
@@ -4989,6 +4990,17 @@
         "node": ">= 4.0.0"
       }
     },
+    "node_modules/atob": {
+      "version": "2.1.2",
+      "resolved": "https://registry.npmjs.org/atob/-/atob-2.1.2.tgz",
+      "integrity": "sha512-Wm6ukoaOGJi/73p/cl2GvLjTI5JM1k/O14isD73YML8StrH/7/lRFgmg8nICZgD3bZZvjwCGxtMOD3wWNAu8cg==",
+      "bin": {
+        "atob": "bin/atob.js"
+      },
+      "engines": {
+        "node": ">= 4.5.0"
+      }
+    },
     "node_modules/autoprefixer": {
       "version": "10.4.13",
       "resolved": "https://registry.npmjs.org/autoprefixer/-/autoprefixer-10.4.13.tgz",
@@ -6442,6 +6454,14 @@
       "resolved": "https://registry.npmjs.org/decimal.js/-/decimal.js-10.4.2.tgz",
       "integrity": "sha512-ic1yEvwT6GuvaYwBLLY6/aFFgjZdySKTE8en/fkU3QICTmRtgtSlFn0u0BXN06InZwtfCelR7j8LRiDI/02iGA=="
     },
+    "node_modules/decode-uri-component": {
+      "version": "0.2.0",
+      "resolved": "https://registry.npmjs.org/decode-uri-component/-/decode-uri-component-0.2.0.tgz",
+      "integrity": "sha512-hjf+xovcEn31w/EUYdTXQh/8smFL/dzYjohQGEIgjyNavaJfBY2p5F527Bo1VPATxv0VYTUC2bOcXvqFwk78Og==",
+      "engines": {
+        "node": ">=0.10"
+      }
+    },
     "node_modules/dedent": {
       "version": "0.7.0",
       "resolved": "https://registry.npmjs.org/dedent/-/dedent-0.7.0.tgz",
@@ -15268,6 +15288,16 @@
         "webpack": "^5.0.0"
       }
     },
+    "node_modules/source-map-resolve": {
+      "version": "0.6.0",
+      "resolved": "https://registry.npmjs.org/source-map-resolve/-/source-map-resolve-0.6.0.tgz",
+      "integrity": "sha512-KXBr9d/fO/bWo97NXsPIAW1bFSBOuCnjbNTBMO7N59hsv5i9yzRDfcYwwt0l04+VqnKC+EwzvJZIP/qkuMgR/w==",
+      "deprecated": "See https://github.com/lydell/source-map-resolve#deprecated",
+      "dependencies": {
+        "atob": "^2.1.2",
+        "decode-uri-component": "^0.2.0"
+      }
+    },
     "node_modules/source-map-support": {
       "version": "0.5.21",
       "resolved": "https://registry.npmjs.org/source-map-support/-/source-map-support-0.5.21.tgz",
@@ -20699,6 +20729,11 @@
       "resolved": "https://registry.npmjs.org/at-least-node/-/at-least-node-1.0.0.tgz",
       "integrity": "sha512-+q/t7Ekv1EDY2l6Gda6LLiX14rU9TV20Wa3ofeQmwPFZbOMo9DXrLbOjFaaclkXKWidIaopwAObQDqwWtGUjqg=="
     },
+    "atob": {
+      "version": "2.1.2",
+      "resolved": "https://registry.npmjs.org/atob/-/atob-2.1.2.tgz",
+      "integrity": "sha512-Wm6ukoaOGJi/73p/cl2GvLjTI5JM1k/O14isD73YML8StrH/7/lRFgmg8nICZgD3bZZvjwCGxtMOD3wWNAu8cg=="
+    },
     "autoprefixer": {
       "version": "10.4.13",
       "resolved": "https://registry.npmjs.org/autoprefixer/-/autoprefixer-10.4.13.tgz",
@@ -21756,6 +21791,11 @@
       "resolved": "https://registry.npmjs.org/decimal.js/-/decimal.js-10.4.2.tgz",
       "integrity": "sha512-ic1yEvwT6GuvaYwBLLY6/aFFgjZdySKTE8en/fkU3QICTmRtgtSlFn0u0BXN06InZwtfCelR7j8LRiDI/02iGA=="
     },
+    "decode-uri-component": {
+      "version": "0.2.0",
+      "resolved": "https://registry.npmjs.org/decode-uri-component/-/decode-uri-component-0.2.0.tgz",
+      "integrity": "sha512-hjf+xovcEn31w/EUYdTXQh/8smFL/dzYjohQGEIgjyNavaJfBY2p5F527Bo1VPATxv0VYTUC2bOcXvqFwk78Og=="
+    },
     "dedent": {
       "version": "0.7.0",
       "resolved": "https://registry.npmjs.org/dedent/-/dedent-0.7.0.tgz",
@@ -27959,6 +27999,15 @@
         "source-map-js": "^1.0.1"
       }
     },
+    "source-map-resolve": {
+      "version": "0.6.0",
+      "resolved": "https://registry.npmjs.org/source-map-resolve/-/source-map-resolve-0.6.0.tgz",
+      "integrity": "sha512-KXBr9d/fO/bWo97NXsPIAW1bFSBOuCnjbNTBMO7N59hsv5i9yzRDfcYwwt0l04+VqnKC+EwzvJZIP/qkuMgR/w==",
+      "requires": {
+        "atob": "^2.1.2",
+        "decode-uri-component": "^0.2.0"
+      }
+    },
     "source-map-support": {
       "version": "0.5.21",
       "resolved": "https://registry.npmjs.org/source-map-support/-/source-map-support-0.5.21.tgz",
diff --git a/package.json b/package.json
index 353e5aa..d6661a6 100644
--- a/package.json
+++ b/package.json
@@ -17,6 +17,7 @@
     "react-router-dom": "^6.4.3",
     "react-scripts": "^5.0.1",
     "react-toastify": "^9.1.1",
+    "source-map-resolve": "^0.6.0",
     "web-vitals": "^2.1.4"
   },
   "scripts": {
diff --git a/src/App.css b/src/App.css
index 69cb803..723d24d 100644
--- a/src/App.css
+++ b/src/App.css
@@ -14,33 +14,38 @@
   align-items: center;
 }
 nav {
-  width: 89.5%;
-  margin-left:23px ;
-  background-color:  rgb(223, 222, 220);
+  width: 100%;
+  /* margin-left:23px ; */
+  background-color: rgb(223, 222, 220);
 }
 .ul {
   width: 30%;
-  height: 5vh;
+  height: 7vh;
   margin-left: 50px !important ;
   display: flex;
   align-items: center;
   justify-content: space-between;
   flex-direction: row;
 }
-.select{
+.ul li:hover {
+  background-color: rgb(139, 136, 136);
+  padding: 1px 6px;
+  color: white;
+}
+.select {
   width: 270px !important;
   padding: 15px;
 }
-.demand-table{
+.demand-table {
   width: 89.6%;
   margin-left: 23px;
 }
-.demand-btn{
+.demand-btn {
   position: absolute;
-  right: 10%;
-  top: 7%;
+  right: 8%;
+  top: 8%;
   /* background-color: rgb(148, 148, 252); */
-  width: 100px;
+  width: 70px;
   text-align: center;
   padding: 10px;
   color: white;
diff --git a/src/App.js b/src/App.js
index 1dd3422..36645a2 100644
--- a/src/App.js
+++ b/src/App.js
@@ -24,6 +24,9 @@ import Header from "./components/Header";
 import DemandPage from "./pages/Demand";
 import DemandPost from "./components/Demand/DemandPost";
 import DemandEdit from "./components/Demand/DemandEdit";
+import Selling from "./pages/Supply";
+import SellingPost from "./components/Supply/SupplyPost"
+import SellingEdit from "./components/Supply/SuppluEdit"
 
 function App() {
   return (
@@ -34,8 +37,11 @@ function App() {
         <Routes>
           <Route path="/" element={<Home />} />
           <Route path="/demand" element={<DemandPage />} />
+          <Route path="/selling" element={<Selling />} />
           <Route path="/demandpost" element={<DemandPost />} />
           <Route path="/demandedit" element={<DemandEdit />} />
+          <Route path="/sellingpost" element={<SellingPost/>}/>
+          <Route path="/sellingedit" element={<SellingEdit/>}/>
         </Routes>
       </BrowserRouter>
     </>
diff --git a/src/components/Demand/DemandEdit.jsx b/src/components/Demand/DemandEdit.jsx
index 9012af8..7649bf5 100644
--- a/src/components/Demand/DemandEdit.jsx
+++ b/src/components/Demand/DemandEdit.jsx
@@ -4,8 +4,8 @@ import Selector from "../filters/DemandDD";
 import SelectPort from "../filters/EditSelect.jsx/SelectPort";
 import SelectGrade from "../filters/EditSelect.jsx/SelectGrade";
 import SelectOrigin from "../filters/EditSelect.jsx/SelectOrigin";
-  import { ToastContainer, toast } from "react-toastify";
-  import { useNavigate } from "react-router-dom";
+import { ToastContainer, toast } from "react-toastify";
+import { useNavigate } from "react-router-dom";
 
 import { FaSearch, FaBars } from "react-icons/fa";
 import axios from "axios";
@@ -16,7 +16,7 @@ function DemandEdit() {
   const location = useLocation();
   const [originId, setOriginId] = useState(null);
   const [gradeId, setgradeID] = useState(null);
-  const navigate=useNavigate()
+  const navigate = useNavigate();
   const [demandValues, setValues] = useState({
     dispatch_qty: "",
     importer: "",
@@ -33,7 +33,6 @@ function DemandEdit() {
     setValues({ ...demandValues, [name]: value });
   }
   function submit() {
-   
     demandValues.grade_id = parseInt(gradeId);
     demandValues.origin_id = parseInt(originId);
     parseInt(demandValues.dispatch_qty);
@@ -69,9 +68,12 @@ function DemandEdit() {
         // let { data, message, status } = res.data;
         console.log("result of creat field", res);
         if (res.status === 200) {
-          alert("ok submited");
-          toast('demand is edited sucessfully!')
-          navigate("/demand");
+          toast.success("demand edited sucessfully!", {
+            position: toast.POSITION.TOP_RIGHT,
+          });
+          setTimeout(() => {
+            navigate("/demand");
+          }, 1500);
         }
       })
       .catch((err) => {
@@ -160,7 +162,7 @@ function DemandEdit() {
       <div className="btn-con">
         <button onClick={submit}> Edit</button>
       </div>
-      <ToastContainer/>
+      <ToastContainer />
     </div>
   );
 }
diff --git a/src/components/Demand/DemandPost.jsx b/src/components/Demand/DemandPost.jsx
index a32fd7d..264f074 100644
--- a/src/components/Demand/DemandPost.jsx
+++ b/src/components/Demand/DemandPost.jsx
@@ -5,12 +5,12 @@ import { FaSearch, FaBars } from "react-icons/fa";
 import axios from "axios";
 import { ToastContainer, toast } from "react-toastify";
 import "react-toastify/dist/ReactToastify.css";
-import {useNavigate} from "react-router-dom"
+import { useNavigate } from "react-router-dom";
 function DemandPost() {
   const [portId, setPortId] = useState(null);
   const [originId, setOriginId] = useState(null);
   const [gradeId, setgradeID] = useState(null);
-  const navigate=useNavigate()
+  const navigate = useNavigate();
   const [demandValues, setValues] = useState({
     dispatch_qty: "",
     importer: "",
@@ -30,28 +30,28 @@ function DemandPost() {
     if (!portId || !originId || !gradeId) {
       toast.error("plz select port id ,grade id ,origin id", {
         position: toast.POSITION.TOP_CENTER,
-        delay:10,
-        autoClose:false
+        delay: 10,
+        autoClose: false,
       });
-    }else{
-    demandValues.grade_id = parseInt(gradeId);
-    demandValues.origin_id = parseInt(originId);
-    parseInt(demandValues.dispatch_qty);
-    parseInt(demandValues.net_physical_stock);
-    demandValues.port_id = parseInt(portId);
-    console.log(demandValues, "demandValues");
-    createDemand(demandValues);
+    } else {
+      demandValues.grade_id = parseInt(gradeId);
+      demandValues.origin_id = parseInt(originId);
+      parseInt(demandValues.dispatch_qty);
+      parseInt(demandValues.net_physical_stock);
+      demandValues.port_id = parseInt(portId);
+      console.log(demandValues, "demandValues");
+      createDemand(demandValues);
 
-    setValues({
-      dispatch_qty: "",
-      importer: "",
-      net_physical_stock: "",
-      demandent_name: "",
-    });
-    setPortId(0);
-    setOriginId(0);
-    setgradeID(0);
-  }
+      setValues({
+        dispatch_qty: "",
+        importer: "",
+        net_physical_stock: "",
+        demandent_name: "",
+      });
+      setPortId(0);
+      setOriginId(0);
+      setgradeID(0);
+    }
   }
 
   const createDemand = (data) => {
@@ -71,7 +71,9 @@ function DemandPost() {
         // let { data, message, status } = res.data;
         console.log("result of creat field", res);
         if (res.status === 200) {
-          navigate('/demand')
+          setTimeout(() => {
+            navigate("/demand");
+          }, 1500);
           toast.success("category created sucessfully!", {
             position: toast.POSITION.TOP_RIGHT,
           });
diff --git a/src/components/Demand/DemandTable.jsx b/src/components/Demand/DemandTable.jsx
index 044bb94..49297c0 100644
--- a/src/components/Demand/DemandTable.jsx
+++ b/src/components/Demand/DemandTable.jsx
@@ -134,7 +134,7 @@ function DemandTable() {
                       {data.importer}
                     </td>
                     <td>
-                      <buton onClick={()=>{navigate("/demandedit", { state: data ,id:data.id});}} className="bg-blue-400 py-2 px-4 rounded-md text-white">
+                      <buton onClick={()=>{navigate("/demandedit", { state: data ,id:data.id});}} className="bg-blue-400 cursor-poiner py-2 px-4 rounded-md text-white">
                         Edit
                       </buton >
                     </td>
diff --git a/src/components/Header.jsx b/src/components/Header.jsx
index 0f82cac..c04e97e 100644
--- a/src/components/Header.jsx
+++ b/src/components/Header.jsx
@@ -13,7 +13,7 @@ function Header() {
             <a href="/demand">Demand</a>
           </li>
           <li>
-            <a href="/">selling</a>
+            <a href="/selling">selling</a>
           </li>
         </ul>
       </nav>
diff --git a/src/style/mainComp.css b/src/style/mainComp.css
index bbc2e59..58f0e0b 100644
--- a/src/style/mainComp.css
+++ b/src/style/mainComp.css
@@ -8,7 +8,7 @@
   margin-left: 10px;
   background-color: rgb(223, 222, 220);
 }
-.demandpost {
+.demandpost,.sellingpost {
   border: 2.2px solid rgb(140, 135, 135);
   margin-left: 20px;
   margin-top: 30px;
@@ -20,7 +20,15 @@
   width: 100%;
   display: grid;
   grid-template-columns: repeat(3, 1fr);
-  height: 40vh;
+  height: 50vh;
+  justify-content: center;
+  align-items: center;
+}
+.sellingpost form {
+  width: 100%;
+  display: grid;
+  grid-template-columns: repeat(3, 1fr);
+  height: 60vh;
   justify-content: center;
   align-items: center;
 }
@@ -32,20 +40,21 @@
   justify-content: center;
   align-items: center;
 }
-.priceform form div ,.demandpost form div {
+.priceform form div ,.demandpost form div ,.sellingpost form div{
   display: flex;
   align-items: center;
   justify-content: center;
   flex-direction: column;
 }
-.priceform form div input ,.demandpost form div input {
+.priceform form div input ,.demandpost form div input ,.sellingpost form div input{
   height: 5vh;
   border-radius: 3px;
   padding: 0px 10px;
 }
-.priceform form div h1 ,.demandpost form div h1{
-  font-size: 25px;
-  font-weight: bold;
+.priceform form div h1 ,.demandpost form div h1,.sellingpost h1{
+  font-size: 17px;
+  font-weight: 600;
+  text-transform: capitalize;
 }
 
 
@@ -55,6 +64,9 @@
   align-items: center;
   justify-content: center;
 }
+.btn-sell button{
+  margin-top: 50px;
+}
 .btn-con button {
   width: 150px;
   background-color: rgb(81, 144, 231);
diff --git a/src/style/sidefilters.css b/src/style/sidefilters.css
index c576baa..92c2469 100644
--- a/src/style/sidefilters.css
+++ b/src/style/sidefilters.css
@@ -14,6 +14,9 @@
   font-weight: bolder;
   font-size: 20px;
 }
+.filterhead  h1{
+  font-size: 16px;
+}
 .search {
   display: block;
   position: relative;
